adapter/src/main/java/org/jvalue/ods/adapterservice/datasource/DatasourceManager.java

```diff
@@ -38,27 +41,25 @@ public Datasource createDatasource(Datasource config) {
     return savedConfig;
   }
 
-  public Optional<Datasource> getDatasource(Long id) {
-    return datasourceRepository.findById(id);
+  public Datasource getDatasource(Long id) throws DatasourceNotFoundException {
+    return datasourceRepository.findById(id).orElseThrow(() -> new DatasourceNotFoundException(id));
   }
 
   public Iterable<Datasource> getAllDatasources() {
     return datasourceRepository.findAll();
   }
 
   @Transactional
-  public void updateDatasource(Long id, Datasource update) throws IllegalArgumentException {
-    Datasource existing = datasourceRepository.findById(id)
-      .orElseThrow(() -> new IllegalArgumentException("Datasource with id " + id + " not found"));
+  public void updateDatasource(Long id, Datasource update) throws DatasourceNotFoundException {
+    Datasource existing = datasourceRepository.findById(id).orElseThrow(() -> new DatasourceNotFoundException(id));
```

Comment:
I think these chains are hard to read.. I guess we should introduce some linter or so in the future xD

Reply:
I actually kinda like chains like that, guess that comes from my preference for functional programming :grin:.